toolchain/gcc.py

```diff
@@ -0,0 +1,103 @@
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2016 IBM.
+# Author: Ramya BS <ramya@linux.vnet.ibm.com>
+
+import os
+import re
+from avocado import Test
+from avocado import main
+from avocado.utils import archive
+from avocado.utils import build
+from avocado.utils import distro
+from avocado.utils import process
+from avocado.utils.software_manager import SoftwareManager
+
+
+class GCC(Test):
+    """
+    This testcase make use of testsuite provided by the
+    source package.
+    """
+
+    def setUp(self):
+        """
+        Install all the dependency packages required for building
+        source tarball specific to os,if not tests will stop.
+        """
+        sm = SoftwareManager()
+        dist = distro.detect()
+        packages = ['gcc', 'dejagnu', 'flex', 'bison', 'sharutils']
+
+        if dist.name == 'Ubuntu':
+            packages.extend(['libmpfr-dev', 'libgmp-dev', 'libmpc-dev',
+                             'texinfo', 'zip', 'libc6-dev', 'libelf1',
+                             'elfutils', 'gnat', 'autogen'])
+        elif dist.name == 'SuSE':
+            packages.extend(['glibc-devel-static', 'zlib-devel', 'elfutils',
+                             'libelf-devel', 'gcc-c++', 'isl-devel',
+                             'gmp-devel', 'glibc-devel', 'mpfr-devel',
+                             'makeinfo', 'texinfo', 'mpc-devel'])
+        else:
+            packages.extend(['glibc-static', 'autogen', 'guile',
+                             'guile-devel', 'libgo', 'libgo-devel',
+                             'libgo-static', 'elfutils-devel',
+                             'texinfo-tex', 'texinfo', 'elfutils-libelf-devel',
+                             'gmp-devel', 'mpfr-devel', 'libmpc-devel',
+                             'gcc-gnat', 'libgnat', 'zlib-devel',
+                             'gettext', 'sharutils', 'libgcj', 'libunwind'
+                             'libgcc', 'libgomp'])
+
+        for package in packages:
+            if not sm.check_installed(package) and not sm.install(package):
+                self.skip(
+                    "Failed to install %s required for this test." % package)
+        if dist.name == 'SuSE':
+            gcc_version = process.system_output("gcc --version")
+            get_system_gcc_version = re.search(
+                'gcc \(SUSE Linux\) (.*)\n',
+                gcc_version).group(1).replace(
+                ".",
+                "_")
+        else:
+            get_system_gcc_version = process.system_output(
+                "gcc -dumpversion").replace(".", "_").split("\n", 1)[0]
+        tarball = self.fetch_asset(
+            "https://github.com/gcc-mirror/gcc/archive/gcc-%s-release.zip" %
+            get_system_gcc_version, expire='7d')
```

Comment:
@ramyabs can you please use upstream git url that can be common for all if you want to make support fedora/centos 


`https://github.com/gcc-mirror/gcc/archive/master.zip ` use this url